portal-web/docroot/html/js/liferay/ddm_form.js

```diff
@@ -282,6 +282,31 @@ AUI.add(
 						);
 					},
 
+					_addFieldValidation: function(newField, originalField) {
+						var instance = this;
+
+						instance.fire(
+							'repeat',
+							{
+								field: newField,
+								originalField: originalField
+							}
+						);
+
+						AArray.each(
+							newField.get('fields'),
+							function(item, index) {
+								var name = item.get('name');
+
+								var originalChildFields = originalField.getFieldsByName(name);
+
+								if (originalChildFields.length) {
+									instance._addFieldValidation(item, originalChildFields[0]);
```

Comment:
Heya @blzaugg, this part is a little weird for me... we're looping with a .filter over an entire collection (which makes sense for a method called `getFieldsByName`, but we're only using the first field?
If we'll only ever need the first field (which I'm assuming is the repeatable field that it's children field repeat from), why not just do an `AArray.find()` and stop looping after we find the first field? We would need to change the method name of course, but if we really only need the parent repeatable, that shouldn't be too hard to name.
Let me know if I'm assuming something incorrect though.
Thanks Byran,